File: 131-PalindromePartitioning/131-PalindromePartitioning.py
# Last updated: 07/04/2026, 11:41:07
class Solution:
    
    def is_pal(self,s,i ,j) :
        while i <= j :
            if s[i]!= s[j] :
                return False

            i+=1
            j-=1
        return True


    def func(self,s,index,path,res) :
        if (index == len(s)):
            res.append(path[:])
            return 

        for i in range(index,len(s)):
            if self.is_pal(s,index,i) :
                path.append(s[index:i+1])
                self.func(s,i+1,path,res)
                path.pop()

            



    
    def partition(self, s: str) -> List[List[str]]:
        path = []
        res=[]
        self.func(s,0,path,res)
        return res



        
        

File: 131-PalindromePartitioning/131-PalindromePartitioning.cpp
// Last updated: 07/04/2026, 11:41:13

class Solution {
public:
    vector<vector<string>> partition(string s) {
        vector < vector <string> > res;
        vector <string> path;
        func(s, 0, path, res);
        return res;
    }


    void func(string s, int index, vector<string>& path, vector<vector<string> >& res) {
        if (index == s.size()) {
            res.push_back(path);
            return;
        }
    


        for (int i = index; i < s.size();i++) {
            
            if (is_pal(s, index, i)) {
                path.push_back(s.substr(index, i - index + 1));
                func(s, i + 1, path, res);
                path.pop_back();

            }


        }
    }




    bool is_pal(string s, int start, int end) {

        while (start <= end) {
            if (s[start++] != s[end--]) {
                return false;
            }
        }
        return true;

    }
};


